File: p2/283/Solution.rs

impl Solution {
    pub fn move_zeroes(nums: &mut Vec<i32>) {
        let mut pos = 0;
        for i in (0..nums.len()) {
            println!("{}",i);
             if (nums[n] != 0) {
                 let mut tmp = nums[n];
                 nums.insert(i, nums[pos]);
                 nums.insert(pos++, tmp);
             }
        }     
    }
}

File: p2/283/Solution.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        pos = 0 
        i = 0
        for n in nums:
            if n != 0:
                nums[pos],nums[i] = nums[i], nums[pos] 
                pos = pos + 1
            i = i + 1